core/src/main/java/com/linecorp/armeria/client/AbstractHttpRequestHandler.java

```diff
@@ -61,6 +65,7 @@ enum State {
         NEEDS_TO_WRITE_FIRST_HEADER,
         NEEDS_DATA,
         NEEDS_DATA_OR_TRAILERS,
+        NEEDS_100_CONTINUE,
```

Comment:
Note: although the RFC seems to elude that we need not wait for the continue response, I think this is fine as a first step.

> since [100 (Continue)](https://www.rfc-editor.org/rfc/rfc9110#status.100) responses cannot be sent through an HTTP/1.0 intermediary, such a client SHOULD NOT wait for an indefinite period before sending the content

ref: https://www.rfc-editor.org/rfc/rfc9110#section-10.1.1-11.3

Reply:
As you mentioned, we should check the protocol version.
So, I added the check in response decoder side.
https://github.com/line/armeria/blob/4c64e9594987d6eddfa277c3d02a6d3f14839b31/core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java#L290-L293

but, I couldn't find any good methods or variables to do it in request side...

Reply:
Armeria is compatible with HTTP/1.0 but we don't fully support it. When sending a request, we assume the protocol is HTTP/1.1 for HTTP 1.

Reply:
> I wasn't suggesting that we check the protocol version. Rather my point was that if a request fails to respond the request should just go through normally. (as if the Expect header didn't exist)

Yeah, it's already handled in this way in this commit: https://github.com/line/armeria/pull/5646/commits/9311e3701e7ef3c1fb56f0e26fbca578b53e321d
https://github.com/line/armeria/pull/5646#discussion_r1606168855